### PR TITLE
Switch to go modules

### LINKED ISSUE #940: Switch to go modules
**Problem**
Knative is planning to switch to go mod (and Go 1.14) soon, and in the process remove dep files (Gopkg.*). This repo will need to switch at roughly the same time to avoid dependency breakages.

**Exit Criteria**
Repo has `go.mod` and no `Gopkg.lock`.

**Additional context (optional)**
See the pinned message in #go-mod in Knative Slack for instructions and  https://github.com/knative/pkg/pull/1179 and https://github.com/knative/eventing/pull/3044 for examples.

### PR DESCRIPTION
**Pull request by ian-mi**
Fixes #940

### CODE REVIEW

File: pkg/broker/handler/pool/fanout/pool_test.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// +build !race
```
[the remark this answers is not in the document]
Comment (ian-mi) [reply pkg/broker/handler/pool/fanout/pool_test.go:1]: The number of goroutines only appears to exceed the limit when the pubsub lib is upgraded. I could upgrade the version before the go module switch if you think it is worth doing that separately.

**Comment by ian-mi** [comment]
The test issues appear to be due to changes in dependencies. I could try pinning all package versions for the switch and unpin them afterwards.

**Comment by ian-mi** [comment]
Actually, it's simply not possible to keep dependency versions. Knative-gcp uses cloud.google.com/go@v0.44.3 which includes a pubsub package. The latest eventing depends on cloud.google.com/go/pubsub which is a new module conflicting with the old pubsub package. This means it is necessary to upgrade to a newer pubsub version.

File: hack/tools.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package tools imports tool dependencies
+package tools
+
+import _ "knative.dev/pkg/hack"
```
Comment (chizhg) [inline hack/tools.go:22]: Please add the imports for https://github.com/google/knative-gcp/blob/master/Gopkg.toml#L7-L11 and https://github.com/google/knative-gcp/blob/master/Gopkg.toml#L20. These are required for testing.

**Comment by ian-mi** [comment]
I've created #983 to update cloud.google.com/go versions to reduce the vendor churn in this PR.

**Comment by ian-mi** [comment]
I've pinned deps to minimize vendor churn. The remaining differences should be due to changes upstream in knative for go modules.

File: pkg/broker/ingress/handler_test.go [hunk 3]
```diff
@@ -159,7 +159,7 @@ func TestHandler(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			ctx := logging.WithLogger(context.Background(), logtest.TestLogger(t))
-			ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+			ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```
[the remark this answers is not in the document]
Comment (ian-mi) [reply pkg/broker/ingress/handler_test.go]: I've reverted all test changes after pinning deps.

File: pkg/broker/handler/pool/fanout/pool_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ian-mi) [reply pkg/broker/handler/pool/fanout/pool_test.go]: Reverted.

**Pull request merged**